lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,33 @@
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict;
+use Mojo::File 'path';
+use Time::Piece;
+
+my $STATE_FILE = "/var/lib/openqa/share/gitlab_maintenance.lock";
+my $MAX_DURATION_HOURS = 24;    #TODO make this a config var ?
+
+sub decide_outcome {
+    my ($app, $ctx, $error_string) = @_;
+
+    unless (-f $STATE_FILE) {
+        eval {
+            path($STATE_FILE)->spurt(time . "\n");    # time of intial failure
```

Comment:
(`spurt` is deprecated and should be replaced by `spew`.)
But `spew` isn't  an atomic operation, so there is a race condition, and the file could end up with partial content.
We would either need file locking, or use a file per repository, because the minion job guards the job with a lock for all paths.

Reply:
Maybe we can just use `touch` and use the file mtime for the timestamp. So we wouldn't need locking.
The race condition between the `-f` and the `touch` wouldn't matter in any case.